src/main/java/com/example/Library/controller/BooksController.java

```diff
@@ -27,22 +27,14 @@ public BooksController(BookService bookService) {
 
     @PreAuthorize("hasAuthority('user:guest')")
     @GetMapping()
-    public ArrayList<Book> getAllBook() {
-        return (ArrayList<Book>) bookService.getAllBooks();
+    public ArrayList<Book> getAllBook(@RequestParam(value = "title",required = false)String title) {
+        return (ArrayList<Book>) bookService.getAllBooksWithFilterByTitle(title);
     }
 
     @PreAuthorize("hasAuthority('user:guest')")
     @GetMapping("/{id}")
-    public Book getBookById(@PathVariable long id) {
-        return bookService.getBookById(id)
-                .orElseThrow(() -> new BookNotFoundByIdException(id));
-    }
-
-    @PreAuthorize("hasAuthority('user:guest')")
-    @GetMapping("/getBookByTitle/{title}")
-    public Book getBookByTitle(@PathVariable String title) {
-        return bookService.getBookByTitle(title)
-                .orElseThrow(() -> new BookNotFoundByTitleException(title));
+    public Optional<Book> getBook(@PathVariable long id) {
```

Comment:
it's not recommended to return Optional in controller response, return Book or throw exception instead (previous implementation was correct)